src/backend/distributed/utils/shardinterval_utils.c

```diff
@@ -74,29 +74,31 @@ CompareShardIntervals(const void *leftElement, const void *rightElement,
 	/*
 	 * Left element should be treated as the greater element in case it doesn't
 	 * have min or max values.
-	 */
-	if (!leftShardInterval->minValueExists || !leftShardInterval->maxValueExists)
-	{
-		comparisonResult = 1;
-		return comparisonResult;
-	}
-
-	/*
 	 * Right element should be treated as the greater element in case it doesn't
 	 * have min or max values.
 	 */
-	if (!rightShardInterval->minValueExists || !rightShardInterval->maxValueExists)
+	if (!leftShardInterval->minValueExists || !leftShardInterval->maxValueExists ||
+		!rightShardInterval->minValueExists || !rightShardInterval->maxValueExists)
 	{
-		comparisonResult = -1;
-		return comparisonResult;
+		comparisonResult = (!leftShardInterval->minValueExists ||
+							!leftShardInterval->maxValueExists) -
+						   (!rightShardInterval->minValueExists ||
+							!rightShardInterval->maxValueExists);
```

Comment:
Is this correct? This returns 0 in cases that two shard intervals aren't equal. For example, when left is `(min=NULL, max=100)` and right is `(min=101, max=NULL)`.

Reply:
Old code would return 1 regardless of which of your examples are left/right. There might be an argument to only compare minValue & not care about maxValue being null. I'm trying to make the most conservative change necessary for my ends